File: solutions/cpp/sieve/1/sieve.cpp
#include "sieve.h"

namespace sieve {

    std::vector<int> primes(const int limit) {
        if (limit < 2) {
            return {};
        }
        std::vector sieve(limit + 1 ,true);
        std::vector<int> result;
        sieve[0] = false;
        sieve[1] = false;
        for (int i = 2; i <= limit; i++) {
            if (sieve[i]) {
                result.push_back(i);
                if (i * i <= limit) {
                    for (int j = i * i; j <= limit; j += i) {
                        sieve[j] = false;
                    }
                }
            }
        }
        return result;
    }

}  // namespace sieve

File: solutions/python/sieve/1/sieve.py
def primes(limit):
    is_composite = []
    result = []
    for _ in range(limit + 1):
        is_composite.append(True)

    for i in range(2,limit+1):
        if is_composite[i]:
            result.append(i)
            for j in range(i * 2, limit + 1, i):
                is_composite[j] = False

    return result
